frontend/tests/integration/components/transportation/tdf/modal-splits-test.js

```diff
@@ -0,0 +1,63 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+module('Integration | Component | transportation/tdf/modal-splits', function(hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  test('temporal split totals calculate correctly when user manually inputs mode split values', async function(assert) {
+    const store = this.owner.lookup('service:store');
+
+    this.server.create('user');
+    this.server.create('data-package');
+    this.server.create('project', {
+      publicSchoolsAnalysis: this.server.create('publicSchoolsAnalysis'),
+      transportationAnalysis: this.server.create('transportationAnalysis', {
+        transportationPlanningFactors: [
+          this.server.create('transportationPlanningFactor', {
+            dataPackage: this.server.create('dataPackage', 'nycAcs'),
+            landUse: 'residential',
+            // necessary for displaying certain elements
+            // am/md/pm/saturday values (true) vs. allPeriods values (false)
+            temporalModeSplits: true,
+            // necessary for displaying certain elements
+            // user input mode splits (true) vs. mode split values from census tract calculator (false)
+            manualModeSplits: true,
+          }),
+        ],
+      }),
+    });
+
+    // define project model
+    const project = await store.findRecord('project', 1, {
+      include: ['transportation-analysis,transportation-analysis.transportation-planning-factors'].join(','),
+    });
+
+    // replicating how availablePackages is defined on routes/project/show/transportation/tdf/planning-factors/show.js
+    const dataPackage = project.transportationAnalysis.get('transportationPlanningFactors').firstObject.get('dataPackage');
+    const availablePackages = await store.query('data-package', {
+      filter: {
+        package: dataPackage.get('package'),
+      },
+    });
+
+    this.set('transportationPlanningFactorResidentialModel', project.transportationAnalysis.get('transportationPlanningFactors').firstObject);
+    this.set('projectModel', project);
+    this.set('transportationAnalysisModel', project.transportationAnalysis);
```

Comment:
One tiny nit that's more stylistic and up to you:

I would try to avoid creating _new_ named variables for properties in the scope. For example, you could do it like this:

```
this.project = project;
```

Then in your template you could say this:
```handlebars
      {{#transportation/tdf/modal-splits
        project=project
        analysis=project.transportationAnalysis
        factor=transportationPlanningFactorResidentialModel
        availablePackages=availablePackages}}
      {{/transportation/tdf/modal-splits}}
```

I get that some things like this are unavoidable, but I suspect you could even get away with taht special model relationship in the template:
```handlebars
factor=project.transportationAnalysis.transportationPlanningFactors.firstObject
```

My reasoning for this is that it creates fewer intermediate steps for a future test reader.

Reply:
Okay makes sense!